lightwood/encoder/numeric/ts_array_numeric.py

```diff
@@ -58,6 +70,16 @@ def decode(self, encoded_values, dependency_data=None, return_all=False):
                                                 self.sub_encoder.output_size)
 
         ret = []
-        for encoded_timestep in torch.split(encoded_values, 1, dim=1):
+        for tensor in torch.split(encoded_values, 1, dim=0):
+            ret.append(self.decode_one(tensor, dependency_data=dependency_data))
+
+        if encoded_values.shape[0] == 1:
+            return ret[0]  # for a single value, we omit wrapping inside a list (TBD in #744)
```

Comment:
No, we shouldn't `decode` and `encode` should have consistent behaviour, always take and return Iterables (well, take Iterables and Tensors respectively, return `List`, but you get the point)

Reply:
Sorry if this requires more annoying changes to usage :(

Reply:
But `if edge case then return changes` is a pattern that gets really weird really quick

Reply:
You're right. Fixed!